src/Symfony/Component/HttpKernel/Controller/ArgumentValueResolver/ArgumentIsRequest.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpKernel\Controller\ArgumentValueResolver;
+
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Controller\ArgumentValueResolverInterface;
+use Symfony\Component\HttpKernel\ControllerMetadata\Argument\ArgumentMetadata;
+
+/**
+ * Supports the same instance as the request object passed along.
+ *
+ * @author Iltar van der Berg <kjarli@gmail.com>
+ */
+final class ArgumentIsRequest implements ArgumentValueResolverInterface
```

Comment:
You are absolutely correct and that's what it was named before. The problem is that "Request" is often typed in the IDE and autocomplete suggests a bunch of alternatives. I often find myself having a list of 5~10 items as suggestions which clutter my original `Request` import. Purely from that speculation I decided to give it a slightly different name.

If you want I can still name it back though.